elm-src/index.elm

```diff
@@ -0,0 +1,489 @@
+module Main exposing (..)
+
+import NixOptions exposing (..)
+import NixJSON exposing (Gson, gnull, gsonDecoder, nixFromGson)
+import Paginate exposing (..)
+import Html exposing (node, Attribute, Html, a, text, em, table, tr, th, td, tbody, thead, input, ul, li, button, h1, h2, div, pre, p, hr)
+import Html.Attributes exposing (placeholder, src, disabled, href, class, autofocus, value, id)
+import Html.Events exposing (onInput, onClick)
+import Http
+import Json.Encode exposing (encode, Value, null)
+import Dict exposing (Dict, empty, foldr, filter)
+import List
+import Navigation
+import QueryString
+import Debug
+import Maybe
+import Array
+import ExEmElm.Parser
+import ExEmElm.Traverse
+import Debounce
+
+
+main =
+    Navigation.program ChangeUrl
+        { init = init
+        , view = view
+        , update = update
+        , subscriptions = subscriptions
+        }
+
+
+
+-- MODEL
+
+
+type alias Model =
+    { query : String
+    , terms : List String
+    , options : NixOptions
+    , matchingOptions : NixOptions
+    , page : Int
+    , location : Navigation.Location
+    , selected : Maybe String
+    , status : PageStatus
+    , debounce_state : Debounce.State
+    }
+
+
+clamped : Model -> Model
+clamped model =
+    { model | page = (clamp 1 (total_pages model.matchingOptions) model.page) }
+
+
+update_options : Model -> NixOptions -> Model
+update_options model unsorted_options =
+    let
+        options =
```

Comment:
General-ish remark: type signatures everywhere prevents a lot of mistakes down the line; even in `let..in` expressions :)

Reply:
No. Way!